llama_stack/distribution/datatypes.py

```diff
@@ -114,7 +114,11 @@ class DistributionSpec(BaseModel):
 class Provider(BaseModel):
     provider_id: str
     provider_type: str
-    config: Dict[str, Any]
+    config: Any
+    provider_path: Optional[str] = Field(
```

Comment:
I think we should do this slightly different. A provider here is an _instance_ -- notice how there  is an `identifier` tied to it. For example, you can have two instances of vLLM going each serving some model. 

What we need is an extension to the `provider_registry`. We need to be able to construct `ProviderSpec` objects from some kind of a config / extension yaml file. Once that is done, everything should work transparently without needing any other changes.

Reply:
Ok, thanks PTAL :)